### PR TITLE
Pruning usage of AttributeVariable and cleaning up associated code

### PR DESCRIPTION
**Pull request by AsherGlick**
We were using the dataclass AttributeVariable very wrongly, this cleans that up by creating a new minimal dataclass for the fields we need to pass into the attribute classes. Additionally cleans up a few other things associated with the code that needed to be changed.

### CODE REVIEW

File: xml_converter/generators/cpp_templates/attribute_template.hpp [hunk 1]
```diff
@@ -15,19 +15,19 @@
     class XMLError;
 
     enum {{class_name}} {
-        {% for attribute_variable in attribute_variables: %}
-        {{attribute_variable.attribute_name}},
+        {% for attribute_variable in attribute_components %}
```
Comment (klingbolt) [inline xml_converter/generators/cpp_templates/attribute_template.hpp:18]: If we are changing this to `attribute_components`, should `attribute_variable` become `attribute_component` as well?

Comment (AsherGlick) [reply xml_converter/generators/cpp_templates/attribute_template.hpp]: Fixed

File: xml_converter/generators/generate_cpp.py [hunk 2]
```diff
@@ -608,6 +565,10 @@ def get_attribute_variable_key(attribute_variable: AttributeVariable) -> str:
     return attribute_variable.attribute_name
```
Comment (klingbolt) [inline xml_converter/generators/generate_cpp.py:565]: You missed this `attribute_variable`

File: xml_converter/generators/generate_cpp.py [hunk 3]
```diff
@@ -573,7 +530,7 @@ def write_attribute(output_directory: str, data: Dict[str, Document]) -> List[st
             template[metadata[filepath]['type']].render(
                 attribute_name=attribute_name,
                 # TODO: Should this attribute_variables list be sorted? The hpp one is.
```
Comment (klingbolt) [inline xml_converter/generators/generate_cpp.py:532]: You missed this `attribute_variable`

Comment (AsherGlick) [reply xml_converter/generators/generate_cpp.py:532]: Fixed

File: xml_converter/generators/generate_cpp.py (hunk 2 above)
Comment (klingbolt) [reply xml_converter/generators/generate_cpp.py:565]: Ignore this.

**Review by klingbolt — APPROVED** [review]

**Pull request merged**